src/main/kotlin/org/jitsi/nlj/rtp/TransportCcEngine.kt

```diff
@@ -74,7 +78,7 @@ class TransportCcEngine(
      * Holds a key value pair of the packet sequence number and an object made
      * up of the packet send time and the packet size.
      */
-    private val sentPacketDetails = LRUCache<Int, PacketDetail>(MAX_OUTGOING_PACKETS_HISTORY)
+    private val sentPacketDetails = ConcurrentSkipListMap<Int, PacketDetail>(::getSequenceNumberDelta)
```

Comment:
Do you think these changes will be valuable permanently?  Or do you think we just need something to convince ourself things are working correctly/debug the current state of things?

Reply:
One reason I ask is that the changes done this way (more permanently) got into the logic quite a bit.  I was wondering before if there was something we could do 'over the top'--just to fix the error logging--which would be less intrusive, but the nature of the problem isn't clear to us right now, I don't think: if we find we've got a bug somewhere we may be able to fix it and reduce/eliminate the scenario causing these problems.  On the other hand, maybe the clients really are sending redundant TCCs a lot in practice and we need to be able to reduce the ambiguity all the time when logging.

Reply:
I do think it's possible that redundant TCCs are being sent; I also think that the LRU cache is the wrong data structure to use for the TCC packet statuses, because old values end up hanging around (that's why we get oldest seq values that don't make much sense).